src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -139,6 +127,22 @@ void symbol_factoryt::gen_nondet_init(
     const pointer_typet &pointer_type=to_pointer_type(type);
     const typet &subtype=ns.follow(pointer_type.subtype());
 
+    if(subtype.id() == ID_struct)
+    {
+      const struct_typet &struct_type = to_struct_type(subtype);
+      const irep_idt struct_tag = struct_type.get_tag();
+
+      if(
+        recursion_set.find(struct_tag) != recursion_set.end() &&
+        depth >= object_factory_params.max_nondet_tree_depth)
```

Comment:
So the `depth` is the minimal length of dereference-chain before null-pointer initialisation? And every chain longer than `depth` will have pointers initialised to null but only when they point to a `struct` that appeared somewhere before on this chain?

Reply:
Yes, `max_nondet_tree_depth` gives the depth of a chain such that a pointer to a struct will be initialised to `null` if the `depth` of the chain exceeds it and the struct appeared already before somewhere on the chain. If there is no recursion the chains can have any depth.

Reply:
I think if there's no recursion then the chains would always have finite length thus it can never be non-terminating. The current behaviour is similar to that of the java object factory.